### DISCUSSION TITLE
"Unknown argument `OR`. Available options are marked with ?."

### DISCUSSION THREAD

**Discussion by juliandelfino** [prisma/prisma#23349] [Q&A]
### Question

im facing some troubles when i want to add a dynamic where, everytime that i add a clause inside an object, it returns 
"Unknown argument `OR`. Available options are marked with ?." 
But if i add it manually to the query it works perfectly, am i missing something?? 

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
model Transactions {
  id                    String     @id @db.VarChar(36) @default(uuid())
  sourceAccountId       String     @db.VarChar(36) 
  sourceAccount         Accounts   @relation("sourceAccount", fields: [sourceAccountId], references: [id])
  destinationAccountId  String     @db.VarChar(36) 
  destinationAccount    Accounts   @relation("destinationAccount", fields: [destinationAccountId], references: [id])
  created_at            DateTime   @db.DateTime(6)
  description           String     @db.VarChar(100)
  amount                Decimal    @db.Decimal(11,2)
}
```
```js
queryObject.where = {
            OR: [
                {
                    sourceAccount: {
                        userId: userId,
                    },
                },
                { destinationAccount: { userId: userId } },
            ],
        };

const [result, total] = await prisma.$transaction([
                prisma.transactions.findMany({
                    take: 20,
                    ...queryObject,
                }),
                prisma.transactions.count(queryObject.where),
            ]);
```

```js
 let transactions, count;

if (!lastRecord) {     
const [result, total] = await prisma.$transaction([
        prisma.transactions.findMany({
                take: 20,
                where: {
                  OR: [
                    {
                      sourceAccount: {
                        userId: "d3356461-3396-4834-8c02-b53808c9cb50"
                      }
                    },
                    {
                      destinationAccount: {
                        userId: "d3356461-3396-4834-8c02-b53808c9cb50"
                      }
                    }
                  ]
                },
                orderBy: {
                  created_at: "desc"
                },
              ? cursor?: TransactionsWhereUniqueInput,
              ? skip?: Int
              })
```

Unknown argument `OR`. Available options are marked with ?.
PrismaClientValidationError:
Invalid `prisma.transactions.findMany()` invocation in

- OS: Windows
- Database:  MySQL,
- Node.js version: v21.1.0

prisma                  : 5.10.2
@prisma/client          : 5.10.2
Computed binaryTarget   : windows
Operating System        : win32
Architecture            : x64
Node.js                 : v21.1.0
Query Engine (Node-API) : libquery-engine 5a9203d0590c951969e85a7d07215503f4672eb9 (at node_modules\.pnpm\@prisma+engines@5.10.2\node_modules\@prisma\engines\query_engine-windows.dll.node)
Schema Engine           : schema-engine-cli 5a9203d0590c951969e85a7d07215503f4672eb9 (at node_modules\.pnpm\@prisma+engines@5.10.2\node_modules\@prisma\engines\schema-engine-windows.exe)
Schema Wasm             : @prisma/prisma-schema-wasm 5.10.0-34.5a9203d0590c951969e85a7d07215503f4672eb9
Default Engines Hash    : 5a9203d0590c951969e85a7d07215503f4672eb9
Studio                  : 0.499.0
Preview Features        : clientExtensions

**Comment by ludralph** [comment]
Hi @juliandelfino 👋 

Thank you for raising this question.

Can you share the `Account` model so I can try reproducing this. Meanwhile, can you try the code below?
```js
const [result, total] = await prisma.$transaction([
  prisma.transactions.findMany({
    take: 20,
    where: {
      OR: [
        {
          sourceAccountId: {
            equals: userId,
          },
        },
        {
          destinationAccountId: {
            equals: userId,
          },
        },
      ],
    },
  }),

```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
